xds/src/test/java/io/grpc/xds/internal/security/trust/XdsX509TrustManagerTest.java

```diff
@@ -537,6 +541,71 @@ public void checkServerTrustedSslEngine()
     assertThat(sslEngine.getSSLParameters().getEndpointIdentificationAlgorithm()).isEmpty();
   }
 
+  @Test
+  public void checkServerTrustedSpiffeTrustBundle()
+      throws CertificateException, IOException, CertStoreException {
+    TestSslEngine sslEngine = buildTrustManagerAndGetSslEngine();
+    X509Certificate[] serverCerts =
+        CertificateUtils.toX509Certificates(TlsTesting.loadCert(SERVER_1_SPIFFE_PEM_FILE));
+    List<X509Certificate> caCerts = Arrays.asList(CertificateUtils
+        .toX509Certificates(TlsTesting.loadCert(CA_PEM_FILE)));
+    trustManager = XdsTrustManagerFactory.createX509TrustManager(
+        ImmutableMap.of("example.com", caCerts), null);
+    trustManager.checkServerTrusted(serverCerts, "ECDHE_ECDSA", sslEngine);
```

Comment:
Please make sure we are running these tests with all 4 checkClient/ServerTrusted methods. For example, it may be good to parametrize the tests to make this happen.

Reply:
The enhancement I made for the `XdsX509TrustManager.check*` methods was changing `delegate` to `chooseDelegate()` - https://github.com/grpc/grpc-java/pull/11627/files/8184839d56d0d2e77befdf58ac737c185f61fc21#diff-fd2e2c88fa0c3a33943fdf235dac294e23ee7ca203db3c94890c653d2d00b923R270

The current test covers a) this change and b) `chooseDelegate()` itself.
Do we really need to create tests for all the other client calls since they'll be testing the same LOCs?

Reply:
I added an extra test for the Server's SslSocket flavor, but, after searching through the codebase, I couldn't find any tests/usages for xDS `checkClientTrusted` other then `checkClientTrusted(X509Certificate[] chain, String authType)` and I already covered it.